recipes/boost/all/conanfile.py

```diff
@@ -871,9 +860,8 @@ def package_info(self):
         if self.options.segmented_stacks:
             self.cpp_info.defines.extend(["BOOST_USE_SEGMENTED_STACKS", "BOOST_USE_UCONTEXT"])
 
-        if self.settings.os != "Android":
```

Comment:
How do you know if this change will not break Android? Did you test, do you have a log?

Reply:
My understanding is that android can only compile with clang and libc++, that's why I believe the condition can be removed.

But, in fact I did not test it neither I have the resources to test it: android dev env, time, etc. My idea was to discover issues during the review, either through the CI or community review.

Could you or do you know who could test this recipe?

Reply:
you can use the Android NDK for testing: https://docs.conan.io/en/latest/integrations/cross_platform/android.html